### PR TITLE
⚡️ Upgrade to Next 12, Remove relay

### PR DESCRIPTION
**Pull request by wei**

### CODE REVIEW

**Comment by wei** [comment]
Removed relay for better performance. It is giving errors which are hard to resolve. If needed, we can add it back fresh in the future.

File: common/configHelper.ts [hunk 1]
```diff
@@ -17,7 +16,7 @@ const DEFAULT_CONFIG: Configuration = {
   pattern: Pattern.plus
 }
 
-const getOptionalConfig = (repository: repoQueryResponse['repository']) => {
+const getOptionalConfig = (repository: any) => {
```
Comment (CryogenicPlanet) [inline common/configHelper.ts:19]: My typesafety nerd really cringes at this tho, we can add apollo client or graphql https://www.the-guild.dev/graphql/codegen

Comment (wei) [reply common/configHelper.ts:19]: hahaha I agree let me see what I can do. I tried using @octokit/graphql.js and even that didn't have a good way to handle type safety on responses.

**Review by CryogenicPlanet — APPROVED** [review]
Mostly looks good to me, except the lack of types on the graph ql stuff.

That said tbh, I feel like this is due for a rewrite to use edge, tailwind, satori (@vercel/og), maybe even the new RSC data fetching stuff.

**Comment by wei** [comment]
Yep I'm looking forward to some of those new features which why I'm getting this upgrade out of the way before we start 😋

**Pull request merged**